src/olgm/gm.js

```diff
@@ -80,39 +82,48 @@ olgm.gm.createLatLng = function(object, opt_ol3map) {
   return new google.maps.LatLng(lonLatCoords[1], lonLatCoords[0]);
 };
 
-
 /**
- * Create a Google Maps LineString or Polygon object using an OpenLayers one.
- * @param {ol.geom.LineString|ol.geom.Polygon} geometry geometry to create
+ * Convert geometry coordinates to latitude and longitude.
+ * @param {Array.<ol.Coordinate>|null} coordinates coordinate to create
  * @param {ol.Map=} opt_ol3map For reprojection purpose. If undefined, then
  *     `EPSG:3857` is used.
- * @return {google.maps.Data.LineString|google.maps.Data.Polygon} google
- * LineString or Polygon
+ * @return {Array<(google.maps.LatLng|google.maps.LatLngLiteral|null)>|null} google LatLng array
  */
-olgm.gm.createGeometry = function(geometry, opt_ol3map) {
+olgm.gm.coordinatesToLatLngs = function(coordinates, opt_ol3map) {
   var inProj = (opt_ol3map !== undefined) ?
-      opt_ol3map.getView().getProjection() : 'EPSG:3857';
-
+  opt_ol3map.getView().getProjection() : 'EPSG:3857';
   var latLngs = [];
   var lonLatCoords;
-
-  var coordinates;
-  if (geometry instanceof ol.geom.LineString) {
-    coordinates = geometry.getCoordinates();
-  } else {
-    coordinates = geometry.getCoordinates()[0];
-  }
-
   for (var i = 0, len = coordinates.length; i < len; i++) {
     lonLatCoords = ol.proj.transform(coordinates[i], inProj, 'EPSG:4326');
     latLngs.push(new google.maps.LatLng(lonLatCoords[1], lonLatCoords[0]));
   }
+  return latLngs;
+};
+
 
+/**
+ * Create a Google Maps LineString or Polygon object using an OpenLayers one.
+ * @param {ol.geom.LineString|ol.geom.MultiLineString|ol.geom.Polygon} geometry geometry to create
+ * @param {ol.Map=} opt_ol3map For reprojection purpose. If undefined, then
+ *     `EPSG:3857` is used.
+ * @return {google.maps.Data.LineString|google.maps.Data.MultiLineString|google.maps.Data.Polygon} google
+ * LineString, MultiLineString or Polygon
+ */
+olgm.gm.createGeometry = function(geometry, opt_ol3map) {
   var gmapGeometry = null;
-  if (geometry instanceof ol.geom.LineString) {
-    gmapGeometry = new google.maps.Data.LineString(latLngs);
+  if (geometry instanceof ol.geom.MultiLineString) {
+    var latLngsArr = [];
+    for (var i = 0; i < geometry.getCoordinates().length; i++) {
```

Comment:
It's faster to do:

```
var coordinates = geometry.getCoordinates();
for (var i = 0, ii  = coordinates.length; i < ii; i++) {
  //
}
```
then you use the `coordinates[i]` instead of calling `getCoordinates()` on each loop iteration.

Please, fix this for the other loops as well.